### PR TITLE
Disable split on numerics for 2-character terms

### PR DESCRIPTION
**Pull request by lovemeblender**

### CODE REVIEW

File: src/main/java/org/skroutz/elasticsearch/index/analysis/WordDelimiterFilter.java [hunk 1]
```diff
@@ -248,7 +248,8 @@ public boolean incrementToken() throws IOException {
         // word of no delimiters, or protected word: just return it
         if ((iterator.current == 0 && iterator.end == termLength) ||
                 (protWords != null && protWords.contains(termBuffer, 0, termLength))
-                || protectedWords.contains(termAttribute.toString())) {
+                || protectedWords.contains(termAttribute.toString())
+                || termAttribute.length() == 2) {
```
Comment (bill-kolokithas) [inline src/main/java/org/skroutz/elasticsearch/index/analysis/WordDelimiterFilter.java]: I suggest putting the length check first for performance reasons.

Comment (lovemeblender) [reply src/main/java/org/skroutz/elasticsearch/index/analysis/WordDelimiterFilter.java]: I agree, thanks!

File: src/main/java/org/skroutz/elasticsearch/index/analysis/WordDelimiterFilter.java [hunk 2]
```diff
@@ -246,9 +246,10 @@ public boolean incrementToken() throws IOException {
         iterator.next();
 
         // word of no delimiters, or protected word: just return it
-        if ((iterator.current == 0 && iterator.end == termLength) ||
-                (protWords != null && protWords.contains(termBuffer, 0, termLength))
-                || protectedWords.contains(termAttribute.toString())) {
+        if ((termAttribute.length() == 2 && isAlphaNumeric(termAttribute.toString())
```
Comment (bill-kolokithas) [inline src/main/java/org/skroutz/elasticsearch/index/analysis/WordDelimiterFilter.java]: Instead of converting `termAttribute` to string, we can get the char array by calling [`buffer()`](https://lucene.apache.org/core/6_5_1/core/org/apache/lucene/analysis/tokenattributes/CharTermAttribute.html#buffer--), this way we don't need to convert again to char array in the `isAlphaNumeric` function.

Even better, we already have the buffer (char array) in the `termBuffer` variable.

File: src/main/java/org/skroutz/elasticsearch/index/analysis/WordDelimiterFilter.java [hunk 3]
```diff
@@ -595,6 +596,17 @@ private boolean has(int flag) {
     return (flags & flag) != 0;
   }
 
+  private boolean isAlphaNumeric(String term) {
+    char[] charArr = term.toCharArray();
+
+    for(char c : charArr) {
+      if(!Character.isLetter(c) && !Character.isDigit(c)) {
```
Comment (bill-kolokithas) [inline src/main/java/org/skroutz/elasticsearch/index/analysis/WordDelimiterFilter.java]: We can combine those checks with [`isLetterOrDigit`](https://docs.oracle.com/javase/7/docs/api/java/lang/Character.html#isLetterOrDigit(char)).

File: src/test/java/org/skroutz/elasticsearch/index/analysis/WordDelimiterTokenFilterFactoryTests.java [hunk 4]
```diff
@@ -170,7 +169,28 @@ public void testPartsAndCatenate() throws IOException {
...
+    Settings indexSettings = Settings.builder()
+            .put(IndexMetaData.SETTING_VERSION_CREATED, Version.CURRENT)
+            .put(Environment.PATH_HOME_SETTING.getKey(), createTempDir().toString())
+            .put("index.analysis.filter.my_word_delimiter.type", TYPE_NAME)
+            .put("index.analysis.filter.my_word_delimiter.split_on_numerics", true)
+            .put("index.analysis.filter.my_word_delimiter.split_on_case_change", "true")
+            .build();
+    TokenFilterFactory filterFactory = filterFactory(indexSettings, FILTER_NAME);
+
+    String alphanum = "b1";
```
Comment (bill-kolokithas) [inline src/test/java/org/skroutz/elasticsearch/index/analysis/WordDelimiterTokenFilterFactoryTests.java]: I would add a test for greek unicode symbols as well, just to be sure the `isLetterOrDigit` properly detects them.
Otherwise we might have to use the [codepoint version](https://docs.oracle.com/javase/7/docs/api/java/lang/Character.html#isLetterOrDigit(int)).

**Review by bill-kolokithas — CHANGES_REQUESTED** [review]

File: src/main/java/org/skroutz/elasticsearch/index/analysis/WordDelimiterFilter.java (hunk 3 above)
Comment (lovemeblender) [reply src/main/java/org/skroutz/elasticsearch/index/analysis/WordDelimiterFilter.java]: Done

File: src/main/java/org/skroutz/elasticsearch/index/analysis/WordDelimiterFilter.java (hunk 2 above)
Comment (lovemeblender) [reply src/main/java/org/skroutz/elasticsearch/index/analysis/WordDelimiterFilter.java]: termBuffer contains a queue of a previously inserted string (something like "bogusTerm" - not sure about its purpose). Therefore I used Arrays.copyOfRange to get a [0, termLength] slice of that.

File: src/test/java/org/skroutz/elasticsearch/index/analysis/WordDelimiterTokenFilterFactoryTests.java (hunk 4 above)
Comment (lovemeblender) [reply src/test/java/org/skroutz/elasticsearch/index/analysis/WordDelimiterTokenFilterFactoryTests.java]: Good idea, added
